packages/core/admin/admin/src/content-manager/components/BlocksEditor/hooks/useBlocksStore.js

```diff
@@ -271,8 +271,18 @@ const Link = React.forwardRef(({ element, children, ...attributes }, forwardedRe
   );
   const [isEditing, setIsEditing] = React.useState(element.url === '');
   const linkRef = React.useRef(null);
-
   const elementText = element.children.map((child) => child.text).join('');
+  const [formData, setFormData] = React.useState({
+    text: elementText,
+    url: element.url,
+  });
+
+  const handleFormDataChange = (e) => {
+    setFormData({
+      ...formData,
+      [e.target.name]: e.target.value,
+    });
+  };
```

Comment:
Calling this formData makes me think this would use the native form data API, which is not the case anymore. Since this now uses React state, I would find it clearer if we had 2 state values, 1 for text and 1 for url. It also removes the need for the complex handleFormDataChange

Reply:
I also had 2 thoughts, and for me having `formState`(formData) is more apt and handleFormDataChange is also not complex. However I don't mind 2nd option as well as there are only 2 fields for now.